### DISCUSSION TITLE
How to set spacing around the '&' on different situation (bit-wise-add and address-of)?

### DISCUSSION THREAD

**Discussion by xiangfeima** [uncrustify/uncrustify#3995] [Q&A]
Hi there,
In C/C++, I want to have space around the bit-wise add operator '&',
meanwhile no space around address-of '&' (say between type-name and var when casting),
it would be something like
bit-wise add:
`(a & b)`
address of when casting:
`(uint8_t*)&c`

would you help to point which option or combination in cfg can do so?

I tried Uncrustify-0.72.0_f with `sp_addr = remove` and `sp_arith = add` but I always get
bit-wise add:
`(a & b)`
address of:
`(uint8_t*) & c`

Thanks a lot.

**Comment by guy-maurel** [comment]
please create an "Issue" with source code, what you expected to get and config file
